# Lets first get our data

In [15]:
!pip install librosa==0.6
!pip install numba==0.48
!pip install tensorflow==1.8
!pip install resampy==0.3.1
#or run requirements.txt

# IN bash: sudo apt-get install libsndfile1
"""
sudo apt update
sudo apt install ffmpeg
"""

  Using cached numba-0.53.1-cp36-cp36m-manylinux2014_x86_64.whl (3.4 MB)
  Using cached llvmlite-0.36.0-cp36-cp36m-manylinux2010_x86_64.whl (25.3 MB)
  Attempting uninstall: llvmlite
    Found existing installation: llvmlite 0.31.0
    Uninstalling llvmlite-0.31.0:
      Successfully uninstalled llvmlite-0.31.0
  Attempting uninstall: numba
    Found existing installation: numba 0.48.0
    Uninstalling numba-0.48.0:
      Successfully uninstalled numba-0.48.0
You should consider upgrading via the '/home/ecbm4040/envTF24/bin/python3 -m pip install --upgrade pip' command.
  Using cached numba-0.48.0-1-cp36-cp36m-manylinux2014_x86_64.whl (3.5 MB)
  Using cached llvmlite-0.31.0-cp36-cp36m-manylinux1_x86_64.whl (20.2 MB)
  Attempting uninstall: llvmlite
    Found existing installation: llvmlite 0.36.0
    Uninstalling llvmlite-0.36.0:
      Successfully uninstalled llvmlite-0.36.0
  Attempting uninstall: numba
    Found existing installation: numba 0.53.1
    Uninstalling numba-0.53.1:
    

'\nreplace "import tensorflow as tf" with\nimport tensorflow.compat.v1 as tf\ntf.disable_v2_behavior()\n'

In [3]:
from urllib.request import urlopen
from io import BytesIO
from zipfile import ZipFile


def download_and_unzip(url, extract_to='.'):
    http_response = urlopen(url)
    zipfile = ZipFile(BytesIO(http_response.read()))
    zipfile.extractall(path=extract_to)
    

download_and_unzip(url = 'http://mirlab.org/dataset/public/MIR-1K.zip',extract_to = './download')

In [5]:
# Get train test split
!python download.py

# Train if Needed
You need to move files around!
You will have to put: train.txt, test.txt, and valid.txt in nndl_SING/data/MIR1K
The Original Unzipped Files (Lyrics, LyricsWav etc) in to the folder: nndl_SING/data/MIR-1K
Notice there are 2 distint folder here!

In [31]:
# 
!python train.py

/home/ecbm4040/envTF24/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:519: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint8 = np.dtype([("qint8", np.int8, 1)])
/home/ecbm4040/envTF24/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:520: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_quint8 = np.dtype([("quint8", np.uint8, 1)])
/home/ecbm4040/envTF24/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:521: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint16 = np.dtype([("qint16", np.int16, 1)])
/home/ecbm4040/envTF24/lib/python3.6/site-packages/tensorflow/python/framework/dty

Step: 1610 Train Loss: 1.865051
Step: 1620 Train Loss: 1.789449
Step: 1630 Train Loss: 1.928974
Step: 1640 Train Loss: 1.793582
Step: 1650 Train Loss: 1.624317
Step: 1660 Train Loss: 1.920552
Step: 1670 Train Loss: 2.162628
Step: 1680 Train Loss: 1.691685
Step: 1690 Train Loss: 1.990360
Step: 1700 Train Loss: 1.685972
Step: 1710 Train Loss: 1.974059
Step: 1720 Train Loss: 1.932527
Step: 1730 Train Loss: 1.726270
Step: 1740 Train Loss: 1.811085
Step: 1750 Train Loss: 2.005909
Step: 1760 Train Loss: 1.444956
Step: 1770 Train Loss: 2.117089
Step: 1780 Train Loss: 1.867877
Step: 1790 Train Loss: 1.976712
Step: 1800 Train Loss: 2.015224
Step: 1800 Validation Loss: 2.628163
Step: 1810 Train Loss: 2.061099
Step: 1820 Train Loss: 1.817364
Step: 1830 Train Loss: 1.938154
Step: 1840 Train Loss: 1.702561
Step: 1850 Train Loss: 1.807152
Step: 1860 Train Loss: 1.884361
Step: 1870 Train Loss: 1.887356
Step: 1880 Train Loss: 1.735094
Step: 1890 Train Loss: 1.925124
Step: 1900 Train Loss: 1.744870
Ste

Step: 3770 Train Loss: 1.265994
Step: 3780 Train Loss: 1.541785
Step: 3790 Train Loss: 1.628910
Step: 3800 Train Loss: 1.279091
Step: 3800 Validation Loss: 2.154243
Step: 3810 Train Loss: 1.646243
Step: 3820 Train Loss: 1.467382
Step: 3830 Train Loss: 1.469779
Step: 3840 Train Loss: 1.470600
Step: 3850 Train Loss: 1.731205
Step: 3860 Train Loss: 1.472014
Step: 3870 Train Loss: 1.789912
Step: 3880 Train Loss: 1.404631
Step: 3890 Train Loss: 1.371904
Step: 3900 Train Loss: 1.529185
Step: 3910 Train Loss: 1.241150
Step: 3920 Train Loss: 1.540258
Step: 3930 Train Loss: 1.369940
Step: 3940 Train Loss: 1.614539
Step: 3950 Train Loss: 1.525716
Step: 3960 Train Loss: 1.248587
Step: 3970 Train Loss: 1.742773
Step: 3980 Train Loss: 1.384659
Step: 3990 Train Loss: 1.630659
Step: 4000 Train Loss: 1.670287
Step: 4000 Validation Loss: 2.917640
Step: 4010 Train Loss: 1.645788
Step: 4020 Train Loss: 1.229547
Step: 4030 Train Loss: 1.135714
Step: 4040 Train Loss: 1.325994
Step: 4050 Train Loss: 1.23420

Step: 5880 Train Loss: 1.053417
Step: 5890 Train Loss: 1.279672
Step: 5900 Train Loss: 1.366638
Step: 5910 Train Loss: 1.141211
Step: 5920 Train Loss: 1.482110
Step: 5930 Train Loss: 1.498683
Step: 5940 Train Loss: 1.223230
Step: 5950 Train Loss: 1.160829
Step: 5960 Train Loss: 1.224482
Step: 5970 Train Loss: 1.201335
Step: 5980 Train Loss: 1.330238
Step: 5990 Train Loss: 1.138924
Step: 6000 Train Loss: 1.331517
Step: 6000 Validation Loss: 1.958453
Step: 6010 Train Loss: 1.482529
Step: 6020 Train Loss: 1.246527
Step: 6030 Train Loss: 1.424652
Step: 6040 Train Loss: 1.196427
Step: 6050 Train Loss: 1.230189
Step: 6060 Train Loss: 1.262683
Step: 6070 Train Loss: 1.373871
Step: 6080 Train Loss: 1.457531
Step: 6090 Train Loss: 1.178789
Step: 6100 Train Loss: 1.319612
Step: 6110 Train Loss: 1.294408
Step: 6120 Train Loss: 1.246949
Step: 6130 Train Loss: 1.279197
Step: 6140 Train Loss: 1.174014
Step: 6150 Train Loss: 1.486188
Step: 6160 Train Loss: 1.285354
Step: 6170 Train Loss: 1.247869
Ste

Step: 8010 Train Loss: 1.094004
Step: 8020 Train Loss: 1.210347
Step: 8030 Train Loss: 1.283330
Step: 8040 Train Loss: 1.266639
Step: 8050 Train Loss: 1.001756
Step: 8060 Train Loss: 1.372735
Step: 8070 Train Loss: 1.045686
Step: 8080 Train Loss: 1.063534
Step: 8090 Train Loss: 1.149076
Step: 8100 Train Loss: 1.431229
Step: 8110 Train Loss: 1.196322
Step: 8120 Train Loss: 1.003397
Step: 8130 Train Loss: 0.989463
Step: 8140 Train Loss: 1.200534
Step: 8150 Train Loss: 1.157094
Step: 8160 Train Loss: 1.156766
Step: 8170 Train Loss: 1.283719
Step: 8180 Train Loss: 1.109174
Step: 8190 Train Loss: 1.261038
Step: 8200 Train Loss: 1.011216
Step: 8200 Validation Loss: 2.642907
Step: 8210 Train Loss: 1.193456
Step: 8220 Train Loss: 1.188620
Step: 8230 Train Loss: 1.007746
Step: 8240 Train Loss: 1.123623
Step: 8250 Train Loss: 0.905853
Step: 8260 Train Loss: 1.222895
Step: 8270 Train Loss: 1.012076
Step: 8280 Train Loss: 1.097430
Step: 8290 Train Loss: 1.075557
Step: 8300 Train Loss: 1.386413
Ste

Step: 10160 Train Loss: 1.329881
Step: 10170 Train Loss: 0.861469
Step: 10180 Train Loss: 1.252404
Step: 10190 Train Loss: 0.932264
Step: 10200 Train Loss: 0.973654
Step: 10200 Validation Loss: 2.625398
Step: 10210 Train Loss: 1.108951
Step: 10220 Train Loss: 1.033662
Step: 10230 Train Loss: 1.001208
Step: 10240 Train Loss: 0.825192
Step: 10250 Train Loss: 1.110257
Step: 10260 Train Loss: 1.115697
Step: 10270 Train Loss: 1.129456
Step: 10280 Train Loss: 0.845266
Step: 10290 Train Loss: 1.088261
Step: 10300 Train Loss: 1.135433
Step: 10310 Train Loss: 1.154914
Step: 10320 Train Loss: 1.012836
Step: 10330 Train Loss: 0.883620
Step: 10340 Train Loss: 0.977711
Step: 10350 Train Loss: 0.918664
Step: 10360 Train Loss: 1.057654
Step: 10370 Train Loss: 0.797629
Step: 10380 Train Loss: 0.966382
Step: 10390 Train Loss: 1.005314
Step: 10400 Train Loss: 1.231537
Step: 10400 Validation Loss: 2.299562
Step: 10410 Train Loss: 0.832516
Step: 10420 Train Loss: 0.786392
Step: 10430 Train Loss: 1.016406


Step: 12210 Train Loss: 0.905944
Step: 12220 Train Loss: 0.850331
Step: 12230 Train Loss: 1.067334
Step: 12240 Train Loss: 0.958705
Step: 12250 Train Loss: 0.984660
Step: 12260 Train Loss: 0.868434
Step: 12270 Train Loss: 0.884067
Step: 12280 Train Loss: 0.762732
Step: 12290 Train Loss: 0.836038
Step: 12300 Train Loss: 0.750202
Step: 12310 Train Loss: 0.946119
Step: 12320 Train Loss: 0.942075
Step: 12330 Train Loss: 0.805907
Step: 12340 Train Loss: 0.963952
Step: 12350 Train Loss: 1.111670
Step: 12360 Train Loss: 0.790703
Step: 12370 Train Loss: 0.888116
Step: 12380 Train Loss: 1.026627
Step: 12390 Train Loss: 0.846633
Step: 12400 Train Loss: 0.915605
Step: 12400 Validation Loss: 2.205266
Step: 12410 Train Loss: 0.898657
Step: 12420 Train Loss: 0.871855
Step: 12430 Train Loss: 0.728906
Step: 12440 Train Loss: 0.741083
Step: 12450 Train Loss: 0.961321
Step: 12460 Train Loss: 0.830610
Step: 12470 Train Loss: 0.968204
Step: 12480 Train Loss: 0.919451
Step: 12490 Train Loss: 0.832026
Step:

Step: 14300 Train Loss: 0.844922
Step: 14310 Train Loss: 0.740525
Step: 14320 Train Loss: 0.878893
Step: 14330 Train Loss: 0.962457
Step: 14340 Train Loss: 0.867303
Step: 14350 Train Loss: 0.779611
Step: 14360 Train Loss: 0.789077
Step: 14370 Train Loss: 0.843503
Step: 14380 Train Loss: 0.901817
Step: 14390 Train Loss: 0.806828
Step: 14400 Train Loss: 0.706466
Step: 14400 Validation Loss: 1.968694
Step: 14410 Train Loss: 0.940099
Step: 14420 Train Loss: 0.653280
Step: 14430 Train Loss: 0.741208
Step: 14440 Train Loss: 0.818459
Step: 14450 Train Loss: 0.868914
Step: 14460 Train Loss: 0.693972
Step: 14470 Train Loss: 0.933930
Step: 14480 Train Loss: 0.750868
Step: 14490 Train Loss: 0.696258
Step: 14500 Train Loss: 0.937413
Step: 14510 Train Loss: 0.788274
Step: 14520 Train Loss: 0.831934
Step: 14530 Train Loss: 0.762043
Step: 14540 Train Loss: 0.688181
Step: 14550 Train Loss: 0.877637
Step: 14560 Train Loss: 0.955303
Step: 14570 Train Loss: 0.937015
Step: 14580 Train Loss: 0.938030
Step:

Step: 16390 Train Loss: 0.799298
Step: 16400 Train Loss: 0.766237
Step: 16400 Validation Loss: 1.530668
Step: 16410 Train Loss: 0.687395
Step: 16420 Train Loss: 0.692860
Step: 16430 Train Loss: 0.774447
Step: 16440 Train Loss: 0.955065
Step: 16450 Train Loss: 0.604759
Step: 16460 Train Loss: 0.724132
Step: 16470 Train Loss: 0.736699
Step: 16480 Train Loss: 0.800859
Step: 16490 Train Loss: 0.731045
Step: 16500 Train Loss: 0.796360
Step: 16510 Train Loss: 0.733004
Step: 16520 Train Loss: 0.796756
Step: 16530 Train Loss: 0.781891
Step: 16540 Train Loss: 0.900717
Step: 16550 Train Loss: 0.739155
Step: 16560 Train Loss: 0.695806
Step: 16570 Train Loss: 0.798087
Step: 16580 Train Loss: 0.799389
Step: 16590 Train Loss: 0.792538
Step: 16600 Train Loss: 0.843092
Step: 16600 Validation Loss: 1.893348
Step: 16610 Train Loss: 0.798763
Step: 16620 Train Loss: 0.778023
Step: 16630 Train Loss: 0.725220
Step: 16640 Train Loss: 0.651024
Step: 16650 Train Loss: 0.656983
Step: 16660 Train Loss: 0.830452


In [28]:
import os
os.getcwd()

'/home/ecbm4040/nndl_SING'

# Evaluate


In [ ]:
!python evaluate.py

# Use to generate splits

In [18]:
!python main.py

/home/ecbm4040/envTF24/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:519: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint8 = np.dtype([("qint8", np.int8, 1)])
/home/ecbm4040/envTF24/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:520: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_quint8 = np.dtype([("quint8", np.uint8, 1)])
/home/ecbm4040/envTF24/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:521: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint16 = np.dtype([("qint16", np.int16, 1)])
/home/ecbm4040/envTF24/lib/python3.6/site-packages/tensorflow/python/framework/dty

In [17]:
import tensorflow as tf
print(tf.__version__)

/home/ecbm4040/envTF24/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:519: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint8 = np.dtype([("qint8", np.int8, 1)])
/home/ecbm4040/envTF24/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:520: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_quint8 = np.dtype([("quint8", np.uint8, 1)])
/home/ecbm4040/envTF24/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:521: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint16 = np.dtype([("qint16", np.int16, 1)])
/home/ecbm4040/envTF24/lib/python3.6/site-packages/tensorflow/python/framework/dty

1.8.0


In [22]:
!pip install pipreqs

  Created wheel for docopt: filename=docopt-0.6.2-py2.py3-none-any.whl size=13724 sha256=a22be73901d7000bb01b5281550942436e4f972dc2c6032ae638313c0346b277
  Stored in directory: /home/ecbm4040/.cache/pip/wheels/3f/2a/fa/4d7a888e69774d5e6e855d190a8a51b357d77cc05eb1c097c9
Successfully built docopt
You should consider upgrading via the '/home/ecbm4040/envTF24/bin/python3 -m pip install --upgrade pip' command.


# Notes:
He trains too long. We will use early stopping as there is not notable improvement past a certain point around past 25k iteration where training loss stagnates between 0.4 and 0.9 and validation loss also stagnats between 1.5 and 2